src/test/kotlin/CrashReaderIntegrationTest.kt

```diff
@@ -1,85 +1,108 @@
 import arrow.core.Either
 import me.urielsalis.mccrashlib.Crash
 import me.urielsalis.mccrashlib.CrashReader
-import me.urielsalis.mccrashlib.parser.ParserError
-import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.ArgumentsProvider
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.junit.jupiter.params.support.AnnotationConsumer
 import java.io.File
 import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Stream
+import kotlin.streams.toList
 
 class CrashReaderIntegrationTest {
-    val crashReader = CrashReader()
-    val tempDir = Files.createTempDirectory("mappings").toFile()
+    @Target(AnnotationTarget.FUNCTION)
+    @Retention(AnnotationRetention.RUNTIME)
+    @ArgumentsSource(ResourceFilesProvider::class)
+    annotation class ResourceFilesSource(val directory: String, val ignoredNameSuffixes: Array<String> = [])
 
-    @Test
-    fun shouldRunAllMinecraftCrashes() {
-        val crashes = File(this::class.java.getResource("crashes/minecraft").file).listFiles()
+    class ResourceFilesProvider : ArgumentsProvider, AnnotationConsumer<ResourceFilesSource> {
+        private lateinit var directory: String
+        private lateinit var ignoredNameSuffixes: Array<String>
 
-        crashes.forEach {
-            val name = it.name
-            val parts = name.split("-")
-            val isModded = parts[0] == "true"
-            val expectedException = parts[1]
+        override fun accept(t: ResourceFilesSource?) {
+            directory = t!!.directory
+            ignoredNameSuffixes = t.ignoredNameSuffixes
+        }
+
+        override fun provideArguments(context: ExtensionContext?): Stream<out Arguments> {
+            val resourceUrl = this::class.java.getResource(directory)
+                ?: throw IllegalArgumentException("Resource directory '$directory' does not exist")
 
-            val crashEither = crashReader.processCrash(it.readLines(), tempDir)
-            assert(crashEither is Either.Right)
-            val crash = (crashEither as Either.Right).b
-            assert(crash is Crash.Minecraft)
-            assert((crash as Crash.Minecraft).modded == isModded)
-            assert(crash.exception.contains(expectedException))
+            return Files.list(Path.of(resourceUrl.toURI()))
```

Comment:
Note: Conversion from `resourceUrl` to URI to `Path` is necessary; the previous code (`File(url.file)`) did not work when the current working directory contained spaces (and probably other non-ASCII chars as well).